.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,24 @@
+# Run secret-dependent integration tests only after /ok-to-test approval
+on:
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - 'README.md'
+      - 'CHANGELOG.md'
+name: unit
+
+jobs:
+  unit:
+    name: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: ./go.mod
+          cache: false
+      - name: Install dependencies
+        run: make tools
+      - name: Create and populate .snowflake/config file
+        run: mkdir $HOME/.snowflake && echo "${{ secrets.SNOWFLAKE_CONFIG_FILE }}" > $HOME/.snowflake/config
```

Comment:
I've seen that GH is replacing the contents with * in output of this action, but it will not prevent someone from reading the created config directly, right? Shouldn't we care about this?

Reply:
I selected the box in our github settings to require approval for all workflows made by outside contributors. So nobody should be able to print this out.

https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks